File: javascript/p0094.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var inorderTraversal = function (root) {
  let result = [];

  function traverse(node) {
    if (!node) return;
    // Traverse the left subtree
    traverse(node.left);
    // Visit the root node
    result.push(node.val);
    // Traverse the right subtree
    traverse(node.right);
  }

  // Start the traversal from the root node
  traverse(root);

  return result;
};

File: python/p0094.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        result = []
        
        def traverse(node: Optional[TreeNode]):
            if node is None:
                return
            # Traverse the left subtree
            traverse(node.left)
            # Visit the root node
            result.append(node.val)
            # Traverse the right subtree
            traverse(node.right)
        
        # Start the traversal from the root node
        traverse(root)
        
        return result
